File: app/services/email_verifier.py
from __future__ import annotations
"""
ZeroBounce Email Verification.

Free tier: 100 free verifications/month.
Checks deliverability for cold email targets.
"""

import httpx
from app.config import get_settings


async def verify_email(email: str) -> dict | None:
    """Verify a single email address. Returns {status, result, score} or None."""
    if not email:
        return None

    api_key = get_settings().zerobounce_api_key
    if not api_key:
        print("ZeroBounce API key not set — skipping verification")
        return None

    try:
        async with httpx.AsyncClient(timeout=15) as client:
            resp = await client.get(
                "https://api.zerobounce.net/v2/validate",
                params={"api_key": api_key, "email": email},
            )
            if resp.status_code != 200:
                return None

            data = resp.json()
            result = "deliverable" if data.get("status") == "valid" else "undeliverable"

            return {
                "status": data.get("status"),
                "result": result,
                "score": 50 if data.get("sub_status") else 100,
            }

    except Exception as e:
        print(f"ZeroBounce verify error for {email}: {e}")
        return None

File: backend/src/services/emailVerifier.js
/**
 * ZeroBounce Email Verification
 *
 * Checks deliverability for cold email targets.
 * Free tier: 100 free verifications/month
 * API docs: https://www.zerobounce.net/docs/
 */

/**
 * Verify a specific email address — checks deliverability.
 * Returns { status, result, score }.
 *
 * status: "valid" | "invalid" | "catch-all" | "unknown" | "spamtrap" | "abuse" | "do_not_mail"
 */
export async function verifyEmail(email) {
  if (!email) return null;

  const zeroBounceKey = process.env.ZEROBOUNCE_API_KEY;
  if (!zeroBounceKey) {
    console.warn('ZeroBounce API key not set — skipping email verification');
    return null;
  }

  try {
    const res = await fetch(`https://api.zerobounce.net/v2/validate?api_key=${zeroBounceKey}&email=${encodeURIComponent(email)}`);
    if (!res.ok) return null;

    const data = await res.json();
    
    // ZeroBounce status values: "valid", "invalid", "catch-all", "unknown", "spamtrap", "abuse", "do_not_mail"
    // Map to simple deliverability result ('deliverable' or 'undeliverable')
    const result = data.status === 'valid' ? 'deliverable' : 'undeliverable';
    
    return {
      status: data.status,
      result: result,
      score: data.sub_status ? 50 : 100,
    };
  } catch (err) {
    console.error(`ZeroBounce verify error for ${email}:`, err.message);
    return null;
  }
}
